src/client/app/actions/barReadings.ts

```diff
@@ -105,25 +106,25 @@ function fetchGroupBarReadings(groupIDs: number[], timeInterval: TimeInterval):
 	return async (dispatch: Dispatch, getState: GetState) => {
 		const barDuration = getState().graph.barDuration;
 		dispatch(requestGroupBarReadings(groupIDs, timeInterval, barDuration));
-		// API expectes a comma-seperated string of IDs
-		const stringifiedIDs = groupIDs.join(',');
 		const readings = await groupsApi.barReadings(groupIDs, timeInterval, barDuration);
 		dispatch(receiveGroupBarReadings(groupIDs, timeInterval, barDuration, readings));
 	};
 }
 
-function fetchMeterCompareReadings(meterIDs: number[], timeInterval: TimeInterval): Thunk {
+function fetchMeterCompareReadings(meterIDs: number[], comparePeriod: ComparePeriod): Thunk {
 	return async (dispatch: Dispatch, getState: GetState) => {
-		const compareDuration = getState().graph.compareDuration;
+		const compareDuration = calculateCompareDuration(comparePeriod);
+		const timeInterval = getState().graph.compareTimeInterval;
 		dispatch(requestMeterBarReadings(meterIDs, timeInterval, compareDuration));
 		const readings = await metersApi.barReadings(meterIDs, timeInterval, compareDuration);
 		dispatch(receiveMeterBarReadings(meterIDs, timeInterval, compareDuration, readings));
 	};
 }
 
-function fetchGroupCompareReadings(groupIDs: number[], timeInterval: TimeInterval) {
+function fetchGroupCompareReadings(groupIDs: number[], comparePeriod: ComparePeriod) {
 	return async (dispatch: Dispatch, getState: GetState) => {
-		const compareDuration = getState().graph.compareDuration;
+		const compareDuration = calculateCompareDuration(comparePeriod);
+		const timeInterval = getState().graph.compareTimeInterval;
 		dispatch(requestGroupBarReadings(groupIDs, timeInterval, compareDuration));
 		const readings = await groupsApi.barReadings(groupIDs, timeInterval, compareDuration);
```

Comment:
I'd eventually want to refactor this out to a `compareApi` type thing, but we can do that later.